### ISSUE TITLE
ktrain distilbert preprocessing fails with SIGKILL

### ISSUE THREAD

**Issue by wDomin** [amaiya/ktrain#435]
Hi,
I'm trying to train a distilbert classifier but it fails on the preprocessing phase when I try to  feed it too much data. I guess the functions text_from_array, text_from_df load everything in memory in order to accomplish the preprocessing and if I don't limit the input x_train/y_train it will slowly fullfil my memory before crashing:
`preprocessing train...
language: en
train sequence lengths:
	mean : 77
	95percentile : 97
	99percentile : 126

Process finished with exit code 137 (interrupted by signal 9: SIGKILL)`

Am I doing something wrong or is this some kind of limitation to the implementation?
Everything works as it should when I work with a subset of my data.

Thx,

**Comment by amaiya** [comment]
Hello:  A number of users have simply trained in segments by resuming training in ktrain.  Neural networks are naturally incremental learners.  For instance, one user wanted to train on 50 million examples but found that training on over 1M resulted in an out-of-memory error.  So, they trained in 50 different segments/batches of 1M documents each by [resuming training](https://github.com/amaiya/ktrain/blob/master/FAQ.md#how-do-i-resume-training-from-a-saved-checkpoint) for each segment.  

Yes, some users have used dask (most often to parallelize the preprocessing of documents).

I should also add that one of the advantages of pretrained models is that you don't need a lot of training examples when fine-tuning.  So, you might want to experiment with fine-tuning on a  sample and see if there is really an accuracy boost with training on the entire (or much larger) dataset.
